source/frontend/src/features/leases/add/ConsultationSubForm.tsx

```diff
@@ -84,4 +56,44 @@ const ConsultationSubForm: React.FunctionComponent<
   );
 };
 
+export const getConsultations = (lease: ApiGen_Concepts_Lease, consultationTypes) => {
+  if (lease.consultations?.length !== consultationTypes.length) {
+    const newConsultations: ApiGen_Concepts_ConsultationLease[] = [];
+
+    consultationTypes.forEach(consultationType => {
+      const newConsultation: ApiGen_Concepts_ConsultationLease = {
+        id: 0,
+        parentLeaseId: lease.id || 0,
+        consultationType: {
+          id: consultationType.id.toString(),
+          description: consultationType.name,
+          displayOrder: null,
+          isDisabled: false,
+        },
+        consultationStatusType: {
+          id: 'UNKNOWN',
```

Comment:
is there a generate code type that we could use here instead of hardcoding? There has been many codetypes changes around lately.

Reply:
I agree, I think making improvements to that is outside the scope of this bug for now, since I'm just moving some existing code around.